server/clients/nostr/channel.go

```diff
@@ -0,0 +1,216 @@
+package nostr
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/nbd-wtf/go-nostr"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+)
+
+type (
+	Channel struct {
+		Name    string `json:"name"`
+		About   string `json:"about"`
+		Picture string `json:"picture"`
+	}
+
+	RelayChannel struct {
+		Channel
+		Relay string `json:"relay"`
+		Id    string `json:"id"`
+	}
+
+	ChannelMessage struct {
+		// Content of the message
+		Content   string `json:"content"`
+		ChannelID string `json:"channel_id"`
+		// MessageID is used for replies
+		MessageID string `json:"message_id"`
+		// PublicKey of author to reply to
+		PublicKey string `json:"public_key"`
+	}
+
+	RelayChannelMessage struct {
+		Content string     `json:"content"`
+		Tags    [][]string `json:"tags"`
+		Relay   string     `json:"relay"`
+		Id      string     `json:"id"`
+	}
+)
+
+const (
+	// kindCreateChannel creates a channel
+	kindCreateChannel = 40
```

Comment:
Is there a reason for those being private?

Reply:
just that i don't have a reason to export them, do i??